src/control/security/grpc_authorization_test.go

```diff
@@ -0,0 +1,92 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package security
+
+import (
+	"testing"
+)
+
+func TestCommonNameToComponent(t *testing.T) {
+	testCases := []struct {
+		testname   string
+		commonname string
+		expected   Component
+	}{
+		{"AdminCN", "admin", ComponentAdmin},
+		{"AgentCN", "agent", ComponentAgent},
+		{"ServerCN", "server", ComponentServer},
+		{"UnknownCN", "knownbadvalue", ComponentUndefined},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.testname, func(t *testing.T) {
+			comp := CommonNameToComponent(tc.commonname)
+			if comp != tc.expected {
+				t.Errorf("result %s; expected %s", comp.String(), tc.expected.String())
+			}
+		})
+	}
+}
+
+func TestHasAccess(t *testing.T) {
+	var negativeCases = 1
+	testCases := []struct {
+		testname string
+		comp     Component
+		method   string
+		expected bool
+	}{
+		{"/ctl.MgmtCtl/StoragePrepare", ComponentAdmin, "/ctl.MgmtCtl/StoragePrepare", true},
```

Comment:
IMO there's no need to repeat the full method name as the test name. Maybe the testname could be shortened.

Reply:
We could just do the function name instead of the full name for the method. That would probably work. Just figured it would be clearer in case of failure.

Reply:
Fixed